### ISSUE TITLE
Documentation for decorators is a bit backwards

### ISSUE THREAD

**Issue by IanVS** [storybookjs/storybook#18896]
**Describe the bug**
I was just looking at https://storybook.js.org/docs/react/writing-stories/decorators#decorator-inheritance, and I wonder if "in the order they are defined" is a bit misleading.  Decorators work a little bit backwards, from the inside out.  The last ones you define will be the first to wrap the story.  So, if you have decorators = [A, B, C], the result is A(B(C(story))).  So, B and A have the result of C, so it's kind of like they "run" backwards.

**Comment by cassus** [comment]
+1 this also confused me

**Comment by srmagura** [comment]
Here's a minimal repro of the issue: https://github.com/srmagura/storybook-decorator-order-repro

```tsx
export default {
  title: "App",
  decorators: [
    (Story) => (
      <div>
        <p>Decorator 1</p>
        <Story />
      </div>
    ),
    (Story) => (
      <div>
        <p>Decorator 2</p>
        <Story />
      </div>
    ),
  ],
} satisfies Meta;

export const Basic: StoryObj = {
  render: () => <p>My story</p>,
};
```

It renders:

![image](https://github.com/storybookjs/storybook/assets/801549/efcc268a-cdbb-4cea-82cc-b15b36d020e1)


FWIW I think the behavior described in the docs is what I would expect, and the actual behavior is surprising.

**Comment by shilman** [comment]
@srmagura I think your repro shows the opposite of @IanVS 's original comment.

If you rewrite it a little bit:

```
const A =  (Story) => (
      <div>
        <p>Decorator 1</p>
        <Story />
      </div>
    );
const B = (Story) => (
      <div>
        <p>Decorator 2</p>
        <Story />
      </div>
    );

export default {
  decorators: [A, B],
};
```

And the result you're seeing is `B(A(story))`.

**Comment by IanVS** [comment]
Yes that is they way they render. I think the docs could just do a better job explaining it, that's all.

**Comment by jonniebigodes** [comment]
@srmagura, if you're interested in helping improve the documentation, I'd be more than welcome to review your contribution. Let me know and we'll go from there. 

Looking forward to hearing from you.

**Comment by srmagura** [comment]
Hey @jonniebigodes, yes, will submit a PR this weekend.

**Comment by jonniebigodes** [comment]
That's great. to hear @srmagura. Looking forward to seeing it.
